app/commands/user/challenges/featured_exercises_progress_12_in_23.rb

```diff
@@ -0,0 +1,49 @@
+class User::Challenges::FeaturedExercisesProgress12In23
+  include Mandate
+
+  initialize_with :user
+
+  def call
+    featured_exercises.filter_map do |exercise_slug, track_slugs|
+      next unless solutions.key?(exercise_slug)
+
+      solved_in_featured_tracks = solutions[exercise_slug].select { |track_slug| track_slugs.include?(track_slug) }
+
+      solved_in_23 = solved_in_featured_tracks.select { |_, year| year == 2023 }
+      next [solved_in_23.keys.first, exercise_slug] if solved_in_23.any?
+
+      solved_before_23 = solved_in_featured_tracks.select { |_, year| year < 2023 }
+      next [solved_before_23.keys.first, exercise_slug] if solved_before_23.size == track_slugs.size
+    end
+  end
+
+  private
+  memoize
+  def solutions
+    user.solutions.published.
+      joins(:track).
+      pluck('exercises.slug', 'tracks.slug', 'solutions.published_at').
+      group_by(&:first).
+      transform_values { |solutions| solutions.map { |solution| [solution[1], solution[2].year] }.to_h }
+  end
+
+  memoize
+  def featured_exercises
+    (FEBRUARY_EXERCISES.map { |e| [e, FEBRUARY_TRACKS] } +
+     MARCH_EXERCISES.map { |e| [e, MARCH_TRACKS] } +
+     APRIL_EXERCISES.map { |e| [e, APRIL_TRACKS] } +
+     MAY_EXERCISES.map { |e| [e, MAY_TRACKS] }).to_h
```

Comment:
```suggestion
    (
    FEBRUARY_EXERCISES.map { |e| [e, FEBRUARY_TRACKS] } +
     MARCH_EXERCISES.map { |e| [e, MARCH_TRACKS] } +
     APRIL_EXERCISES.map { |e| [e, APRIL_TRACKS] } +
     MAY_EXERCISES.map { |e| [e, MAY_TRACKS] }
     ).to_h
```

(With correct indenting)